library/src/main/java/org/apache/apex/malhar/lib/state/spillable/SpillableStateStore.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.malhar.lib.state.spillable;
+
+import org.apache.apex.malhar.lib.state.BucketedState;
+
+import com.datatorrent.api.Component;
+import com.datatorrent.api.Context;
+import com.datatorrent.api.Operator;
+
+public interface SpillableStateStore extends BucketedState, Component<Context.OperatorContext>,
```

Comment:
Should we make this more generic and have a separate interface called WindowListener that is basically:
```java
interface WindowListener
{
  void beginWindow(long windowId);
  void endWindow();
}
```
and have SpillableStateStore extends WindowListener, BucketedState, and Component<Context.OperatorContext>?
The reason is the operator code will have to call these beginWindow and endWindow callbacks and it's better to make this as generic as possible.

For example, I would have something like this in the operator code:
```java
class SomeOperator extends BaseOperator
{
  List<Component> components;
  
  @Override
  void setup(OperatorContext context)
  {
    for (Component c : components) {
      c.setup(context);
    }
  }

  @Override
  void beginWindow(long windowId)
  {
    for (Component c : components) {
      if (c instanceof WindowListener) {
        ((WindowListener)c).beginWindow(windowId);
      }
    }
  }
  // similar for endWindow and CheckpointListener callbacks
}
```